pkg/utils/utils_test.go

```diff
@@ -421,3 +713,2142 @@ func TestSetContainerImage(t *testing.T) {
 		})
 	}
 }
+
+func TestGetBackupStorageLocation(t *testing.T) {
+	ctx := context.Background()
+	fakeClient := fullFakeClient()
+
+	// Test case: BackupStorageLocation does not exist
+	name := "test-backup-storage"
+	namespace := "test-namespace"
+	_, err := GetBackupStorageLocation(ctx, name, namespace, fakeClient)
+	if err == nil {
+		t.Errorf("Expected error, got nil")
+	}
+
+	// Test case: BackupStorageLocation exists
+	backupStorage := &velerov1.BackupStorageLocation{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+	if err := fakeClient.Create(ctx, backupStorage); err != nil {
+		t.Errorf("Failed to create BackupStorageLocation: %v", err)
+	}
+	backupStorage, err = GetBackupStorageLocation(ctx, name, namespace, fakeClient)
+	if err != nil {
+		t.Errorf("Failed to get BackupStorageLocation: %v", err)
+	}
+	if backupStorage.Name != name {
+		t.Errorf("Expected name %s, got %s", name, backupStorage.Name)
+	}
+	if backupStorage.Namespace != namespace {
+		t.Errorf("Expected namespace %s, got %s", namespace, backupStorage.Namespace)
+	}
+}
+
+func TestUpdateSideCarApply(t *testing.T) {
+	// Test case: update sidecar with matching name
+	sc1env1 := "sidecar1-env1"
+	sc1env2 := "sidecar1-env2"
+	sc1env3 := "sidecar1-env3"
+	oldenv1val := "old-env1-value"
+	oldenv3val := "old-env3-value"
+	newenv1val := "sidecar1-env1-value"
+	newenv2val := "sidecar1-env2-value"
+	sideCars := []csmv1.ContainerTemplate{
+		{
+			Name:            "sidecar1",
+			Image:           "sidecar1-image",
+			ImagePullPolicy: "sidecar1-image-pull-policy",
+			Envs: []corev1.EnvVar{
+				{
+					Name:  sc1env1,
+					Value: newenv1val,
+				},
+				{
+					Name:  sc1env2,
+					Value: newenv2val,
+				},
+			},
+		},
+		{
+			Name:            "sidecar2",
+			Image:           "sidecar2-image",
+			ImagePullPolicy: "sidecar2-image-pull-policy",
+			Envs: []corev1.EnvVar{
+				{
+					Name:  "sidecar2-env1",
+					Value: "sidecar2-env1-value",
+				},
+				{
+					Name:  "sidecar2-env2",
+					Value: "sidecar2-env2-value",
+				},
+			},
+		},
+	}
+
+	container := acorev1.Container().
+		WithName("sidecar1").
+		WithImage("old-image").
+		WithImagePullPolicy("old-image-pull-policy").
+		WithEnv(&acorev1.EnvVarApplyConfiguration{
+			Name:  &sc1env1,
+			Value: &oldenv1val,
+		},
+		).WithEnv(&acorev1.EnvVarApplyConfiguration{
+		Name:  &sc1env3,
+		Value: &oldenv3val,
+	},
+	)
+
+	UpdateSideCarApply(sideCars, container)
+
+	expectedContainer := acorev1.Container().WithName("sidecar1").WithImage("sidecar1-image").WithImagePullPolicy("sidecar1-image-pull-policy").
+		WithEnv(&acorev1.EnvVarApplyConfiguration{
+			Name:  &sc1env1,
+			Value: &newenv1val,
+		}).
+		/*WithEnv(&acorev1.EnvVarApplyConfiguration{ // IF we want to have new vars added in the Apply method, this will need to be uncommented.
```

Comment:
Can this comment be removed?

Reply:
Yes, but I think we should discuss the Apply method first-- it may be functionally broken. 

Apply does not apply any new env variables. It only updates existing ones. If there are some new ones in the input to Apply, they are totally ignored instead of added. If some are NOT in the input and ARE on the existing object, they're also ignored instead of deleted. 

This behavior is not consistent with things like `kubectl apply` and may be a bug, so we should talk about it and see if we WANT Apply to only affect objects already on the target object.

Reply:
After discussion with Shayna, it's likely that our best course of action is to leave this as a TODO and address it when we're in a position to run all e2e tests. Making the change right now would require a dramatic amount of testing to prove that nothing was broken by it, and we are currently blocking other PRs.